drivers: i2c: Enable Gecko i2c driver for efr32bg_sltb010a

fkokosinski opened the pull request:
This PR enables the Gecko i2c driver on the efr32bg_sltb010a board.

msierszulski committed:
drivers: i2c: Enable Gecko i2c driver for efr32bg_sltb010a

This commit enables the Gecko i2c driver on the efr32bg_sltb010a board.

Signed-off-by: Mateusz Sierszulski <msierszulski@antmicro.com>

kgugala reviewed (dismissed):
LGTM

MaureenHelm commented on boards/arm/efr32bg_sltb010a/efr32bg_sltb010a.dts:
```diff
@@ -115,3 +115,9 @@
 };
+
+&i2c0 {
+	location-sda= <GECKO_LOCATION(3) GECKO_PORT_D GECKO_PIN(2)>;
+	location-scl= <GECKO_LOCATION(3) GECKO_PORT_D GECKO_PIN(3)>;
```
Please fix the formatting
```suggestion
	location-sda = <GECKO_LOCATION(3) GECKO_PORT_D GECKO_PIN(2)>;
	location-scl = <GECKO_LOCATION(3) GECKO_PORT_D GECKO_PIN(3)>;
```

lpawelcz replied:
Fixed

MaureenHelm reviewed (approved):
You may also want to update the board yaml and doc to reflect the added I2C support

MaureenHelm reviewed (changes requested):
Sorry, clicked the wrong button in my review

lpawelcz committed:
boards: arm: efr32bg: list I2C in documentation

Add entry for I2C in efr32bg_sltb010a documentation and twister metadata
yaml file.

Signed-off-by: Pawel Czarnecki <pczarnecki@antmicro.com>

lpawelcz committed:
soc: arm: efr32bg22: defconfig: remove I2C_GECKO default n

Don't disable I2C by default

Signed-off-by: Pawel Czarnecki <pczarnecki@antmicro.com>

lpawelcz committed:
soc: arm: efr32bg22: include soc_gpio.h

Include header required in Gecko I2C driver

Signed-off-by: Pawel Czarnecki <pczarnecki@antmicro.com>

carlescufi commented:
@fkokosinski please rebase

lpawelcz commented:
@carlescufi I rebased this PR in order to solve conflict created by merging https://github.com/zephyrproject-rtos/zephyr/pull/53245.
@MaureenHelm please take another look at this PR
